dex/farm/src/lib.rs

```diff
@@ -613,4 +613,214 @@ pub trait Farm:
             &context.get_input_attributes().unwrap().compounded_reward,
         )
     }
+
+    //This function is neccessary in old farms.
+    #[payable("*")]
+    #[endpoint(migrateToNewFarm)]
+    fn migrate_to_new_farm(
+        &self,
+        #[var_args] orig_caller_opt: OptionalArg<ManagedAddress>,
+    ) -> EsdtTokenPayment<Self::Api> {
+        assert!(self, !self.farm_configuration().is_empty(), b"empty config");
+        let config = self.farm_configuration().get();
+        assert!(self, config.is_old, b"bad config");
+
+        let new_farm_address = config.new_farm_address;
+        let mut context = self.new_farm_context(OptionalArg::None);
+
+        // Same as Exit Farm, since we want to migrate the rewards and compounded tokens also
+        {
+            self.load_state(&mut context);
+            assert!(
+                self,
+                context.get_contract_state().unwrap() == &State::Migrate,
+                ERROR_NOT_MIGRATION
+            );
+
+            self.load_farm_token_id(&mut context);
+            assert!(
+                self,
+                !context.get_farm_token_id().unwrap().is_empty(),
+                ERROR_NO_FARM_TOKEN,
+            );
+
+            self.load_farming_token_id(&mut context);
+            assert!(self, context.is_accepted_payment_exit(), ERROR_BAD_PAYMENTS,);
+
+            self.load_reward_token_id(&mut context);
+            self.load_reward_reserve(&mut context);
+            self.load_block_nonce(&mut context);
+            self.load_block_epoch(&mut context);
+            self.load_reward_per_share(&mut context);
+            self.load_farm_token_supply(&mut context);
+            self.load_division_safety_constant(&mut context);
+            self.load_farm_attributes(&mut context);
+
+            self.generate_aggregated_rewards(context.get_storage_cache_mut());
+            self.calculate_reward(&mut context);
+            context.decrease_reward_reserve();
+            self.calculate_initial_farming_amount(&mut context);
+            self.increase_reward_with_compounded_rewards(&mut context);
+            self.commit_changes(&context);
+
+            // Update the farm supply even though we dont burn the tokens by ourselves.
+            // The position will be burned and this supply has to be updated.
+            self.farm_token_supply()
+                .update(|x| *x -= &context.get_tx_input().get_payments().get_first().amount);
+        }
+
+        let mut payments = ManagedVec::<Self::Api, EsdtTokenPayment<Self::Api>>::new();
+        payments.push(context.get_tx_input().get_payments().get_first().clone());
+        payments.push(context.get_final_reward().unwrap().clone());
```

Comment:
When you are migrating from old farm to new farm it is supposed to also send the actual farming tokens. Does these two payments includes also the actual farming tokens?

Reply:
they do not. as stated, they contain the farm token and the reward token